pkg/controller/persistentvolume/index.go

```diff
@@ -87,15 +89,29 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *api.PersistentVo
 		unboundVolumes := []*api.PersistentVolume{}
 		for _, volume := range volumes {
 			// volume isn't currently bound or pre-bound.
-			if volume.Spec.ClaimRef == nil {
-				unboundVolumes = append(unboundVolumes, volume)
-				continue
-			}
-
 			if isVolumeBoundToClaim(volume, claim) {
```

Comment:
@jsafrane Interestingly, the only thing that will break this loop is if you find a volume which is alreay bound from the volume to the claim.  Specifying `volumeName` doesn't affect anything.  Seems like we have run into some ambiguity here.

I think the ambiguity is handled in an outer level.  The calling code only attempts to find the right match if `volumeName` isn't set, so we don't need to handle it in this code.  I'll add a comment explaining it in the code, and we should definitely clarify it in the API.  I am going to punt on that for a bit while I get the code right and think about how much detail to go into.  Since the semantics aren't really plumbed through to the API, I'm not sure exactly the right way to explain this bit without explaining the whole thing together.  That being the case, I'd like to clarify the semantics in a follow-up.